sweetest/sweetest-core/src/main/java/com/mysugr/sweetest/framework/dependency/DependencyState.kt

```diff
@@ -95,74 +92,87 @@ internal class DependencyState<T : Any>(
-                        "auto-initialized. Please define how to instantiate it by adding a `provide { ... }` " +
-                        "configuration!"
-                )
-            }
-            if (constructors.size > 1) {
-                throw IllegalArgumentException(
-                    "Dependencies like \"${configuration.clazz.simpleName}\" which have more than one constructor " +
-                        "can't be auto-initialized. Please define how to instantiate it by adding a " +
-                        "`provide { ... }` configuration!"
-                )
-            }
-            val constructor = constructors.first()
-            val argumentTypes = constructor.parameters.map { it.type.classifier as KClass<*> }
-
-            val arguments = try {
-                argumentTypes.map { TestEnvironment.dependencies.getDependencyState(it).instance }.toTypedArray()
-            } catch (exception: Exception) {
-                throw RuntimeException(
-                    "At least one dependency required by the constructor could " +
-                        "not be found.",
-                    exception
-                )
-            }
+            val constructor = getConstructor()
+            val constructorArguments = getConstructorArguments(constructor)
+            constructor.call(*constructorArguments)
+        } catch (exception: Exception) {
+            throw RuntimeException(
+                "Couldn't automatically construct instance of dependency \"$configuration\"",
+                exception
+            )
+        }
+    }
 
-            constructor.call(*arguments)
+    private fun getConstructorArguments(constructor: KFunction<T>): Array<Any> {
+        return constructor.getParameterTypes()
+            .map(::getSubDependencyInstanceOfType)
+            .toTypedArray()
+    }
+
+    private fun KFunction<T>.getParameterTypes() =
+        this.parameters.map { it.type.classifier as KClass<*> }
+
+    private fun getConstructor(): KFunction<T> {
+        val constructors = configuration.clazz.constructors
+        if (configuration.clazz.isAbstract) {
+            throw IllegalArgumentException(
+                "Dependencies like \"${configuration.clazz.simpleName}\" which are abstract can not be " +
+                    "auto-provided. Please define how to instantiate it by adding a " +
+                    "`provide<${configuration.clazz.simpleName}> { ... }` configuration!"
+            )
+        }
+        if (constructors.size > 1) {
+            throw IllegalArgumentException(
+                "Dependencies like \"${configuration.clazz.simpleName}\" which have more than one constructor " +
+                    "can't be auto-provided. Please define how to instantiate it by adding a " +
+                    "`provide<${configuration.clazz.simpleName}> { ... }` configuration!"
+            )
+        }
+        val constructor = constructors.first()
```

Comment:
A thing that came to my mind: If there are multiple candidates, it might makes sense to take the one with an `@Inject` annotation 🤔 
(just a thought, no need to change)

Reply:
very good idea 👌 i put this to the ideation github project!